### PR TITLE
fix(batchPriority) check out of bounds priority

### PR DESCRIPTION
**Pull request by jvancoillie**
| Q                | A
| ---------------- | ---
| PHP version?     | 7.4
| Bundle version?  | next (as patch)
| New feature?     | no
| Bug fix?         | yes
| Discussion?      | #79

### CODE REVIEW

**Comment by jvancoillie** [comment]
maybe you prefer that we add this check in the setter of the task

**Comment by Guikingone** [comment]
@jvancoillie Yes, always in the task class if possible 🙂

**Comment by jvancoillie** [comment]
I understand I will correct that. must we keep the constants

File: src/Task/AbstractTask.php [hunk 1]
```diff
@@ -25,6 +25,9 @@
  */
 abstract class AbstractTask implements TaskInterface
 {
+    public const MIN_PRIORITY = -1000;
+    public const MAX_PRIORITY  = 1000;
+
```
Comment (Guikingone) [inline src/Task/AbstractTask.php:30]: If we don't use the const outside of the class, consider using `private` as visibility.

File: src/Task/AbstractTask.php [hunk 2]
```diff
@@ -717,11 +718,11 @@ private function validateNice(int $nice = null): bool
 
     private function validatePriority(int $priority): bool
     {
-        if ($priority > 1000) {
+        if ($priority > self::MAX_PRIORITY) {
             return false;
         }
 
-        return $priority >= -1000;
+        return $priority >= self::MIN_PRIORITY;
```
Comment (Guikingone) [inline src/Task/AbstractTask.php:725]: What if we use a ternary?

File: tests/Task/AbstractTaskTest.php [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+    public function testTaskPrioritySetter(): void
+    {
+        $abstractTask = $this->getMockForAbstractClass(AbstractTask::class, ['name' => 'foo']);
+
+        self::assertEquals(1000, $abstractTask->setPriority(1000)->getPriority());
+        self::assertEquals(1000, $abstractTask->setPriority(1001)->getPriority());
+        self::assertEquals(-1000, $abstractTask->setPriority(-1000)->getPriority());
+        self::assertEquals(-1000, $abstractTask->setPriority(-1001)->getPriority());
+        self::assertEquals(5, $abstractTask->setPriority(5)->getPriority());
+    }
```
Comment (Guikingone) [inline tests/Task/AbstractTaskTest.php:24]: Consider adding these tests inside the `NullTaskTest` class, I'm not a big fan of using `getMockForAbstractClass` and `AbstractTask` is never instantiated so there's no benefice in my opinion to add a test class for it.

Comment (jvancoillie) [reply tests/Task/AbstractTaskTest.php:24]: ok i understand, i didn't want the `NullTask` to interfere with any options. I wanted to make sure to test the master method. i will change that

File: src/Task/AbstractTask.php (hunk 1 above)
Comment (jvancoillie) [reply src/Task/AbstractTask.php:30]: yes, I didn't know if it would be useful to use it outside. if it happens, we will be able to use it in public, you are right

File: src/Task/AbstractTask.php (hunk 2 above)
Comment (jvancoillie) [reply src/Task/AbstractTask.php:725]: how? and how will rector react?

Comment (Guikingone) [reply src/Task/AbstractTask.php:725]: Rector has been removed. 

Could be: 

```php
return $priority > self::MAX_PRIORITY ? false : $priority >= self::MIN_PRIORITY;
```

?

**Comment by Guikingone** [comment]
If possible, could you improve the `BatchPolicyTest` to add tests for negative priority? Thanks 🙂

**Comment by jvancoillie** [comment]
Ok I will add tests on the Policy batch.

In the bundle documentation I see that the sort is ascending. 
Therefore the lower the number, the earlier a task is executed in reverse of how the priorities work in symfony.

you confirm that this is what is expected ? 
If so, currently I think the sorting is working backwards for the batchPriority.

**Comment by Guikingone** [comment]
> Therefore the lower the number, the earlier a task is executed in reverse of how the priorities work in symfony.

Yes, with time, the priority will grown and the task will be executed later.

**Comment by jvancoillie** [comment]
@Guikingone tell me if things need to be improved for me it's ok. I see you are waiting for this correction

**Comment by Guikingone** [comment]
Reviewed, looks good to me 🙂

**Comment by Guikingone** [comment]
Thanks for this patch @jvancoillie, your contribution is highly appreciated, sorry for the bug 🙁 

Should be released in `0.4.9` 🙂
